.github/workflows/build-test.yml

```diff
@@ -28,6 +28,8 @@ jobs:
           check-latest: true
       - name: Test source headers are present
         run: make test-source-headers
+      - name: Check if go modules need to be tidied
+        run: go mod tidy -diff
```

Comment:
since you also added `tidy` on the `build` target, it will automatically run before build when we call the `Test build` step below. so we don't need this here now, right?
```
build: $(BINDIR)/$(BINNAME) tidy
```

Reply:
`go mod tidy -diff` checks if the user ran `go mod tidy`.

`build: $(BINDIR)/$(BINNAME) tidy` is a shortcut to ensure users don't forget to tidy up

Reply:
oh thank you, i read that too fast. lgtm :+1: